### PR TITLE
[POAE7-2784] Some improvements and clean for code

### PR DESCRIPTION
**Pull request by Deegue**
### What changes were proposed in this pull request?
1. One to one batch checker uses `memcmp` and cannot be ignore order.
2. Allocated memory of `Cells` should be initialized for `isZero()`, which is the same as it is in `FixedHashTable`.
3. Remove some parts we don't need anymore.
4. Fix code conflict of CI for https://github.com/intel/BDTK/pull/317

### Why are the changes needed?
Improvements.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
UTs.

### CODE REVIEW

File: src/cider/tests/utils/CiderBatchChecker.cpp [hunk 1]
```diff
@@ -382,7 +382,7 @@ bool CiderBatchChecker::checkArrowEq(
   // This check is only a shortcut
   // even if it fails, it does not necessarily mean the results are wrong
   // we can goto step 4 and check the results row-by-row
-  if (expected_batches.size() == 1 && actual_batches.size() == 1) {
+  if (!ignore_order && expected_batches.size() == 1 && actual_batches.size() == 1) {
```
Comment (kd97) [inline src/cider/tests/utils/CiderBatchChecker.cpp:385]: ignore_order will make step3 fail but go through step4? so this is just for skipping useless steps?

File: src/cider/exec/template/IRCodegen.cpp [hunk 2]
```diff
@@ -266,18 +266,16 @@ std::unique_ptr<CodegenColValues> CodeGenerator::codegenConstantExpr(
...
       UNREACHABLE();
     default:
       CHECK_EQ(constant_value.size(), 1);
       return std::make_unique<FixedSizeColValues>(
           constant_value[0],
-          constant_expr->get_is_null()
-              ? llvm::ConstantInt::getTrue(cgen_state_->context_)
-              : llvm::ConstantInt::getFalse(cgen_state_->context_));
+          constant_expr->get_is_null() ? llvm::ConstantInt::getTrue(cgen_state_->context_)
+                                       : nullptr);
```
Comment (kd97) [inline src/cider/exec/template/IRCodegen.cpp:278]: seems old code path, need to change for nextgen?

File: src/cider/tests/utils/CiderBatchChecker.cpp (hunk 1 above)
Comment (Deegue) [reply src/cider/tests/utils/CiderBatchChecker.cpp:385]: Yes, it won't affect correctness, but the logic here should be more precise.

File: src/cider/exec/template/IRCodegen.cpp (hunk 2 above)
Comment (Deegue) [reply src/cider/exec/template/IRCodegen.cpp:278]: Yes, am on it, also get some uts failed.

Comment (Deegue) [reply src/cider/exec/template/IRCodegen.cpp:278]: Changing this leads to some other errors for handling the nullptr.

Comment (Deegue) [reply src/cider/exec/template/IRCodegen.cpp:278]: Reverted this part.

**Review by kd97 — DISMISSED** [review]

**Review by YantingTao1315 — APPROVED** [review]

**Review by kd97 — APPROVED** [review]

**Review by bigPYJ1151 — APPROVED** [review]

**Pull request merged**